src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Sorter/Attributes/Metric/MetricSorterIntegration.php

```diff
@@ -0,0 +1,70 @@
+<?php
+use Pim\Bundle\CatalogBundle\tests\integration\PQB\AbstractProductQueryBuilderTestCase;
+use Pim\Component\Catalog\Query\Sorter\Directions;
+
+/**
+ * Metric sorter integration tests
+ *
+ * @author    Samir Boulil <samir.boulil@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class MetricSorterIntegration extends AbstractProductQueryBuilderTestCase
+{
+    /**
+     * @{@inheritdoc}
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        if (1 === self::$count || $this->getConfiguration()->isDatabasePurgedForEachTest()) {
+            $this->createProduct('product_one', [
+                'values' => [
+                    'a_metric' => [
+                        ['data' => ['amount' => '10.55', 'unit' => 'KILOWATT'], 'locale' => null, 'scope' => null]
+                    ]
+                ]
+            ]);
+
+            $this->createProduct('product_two', [
+                'values' => [
+                    'a_metric' => [
+                        ['data' => ['amount' => '15', 'unit' => 'KILOWATT'], 'locale' => null, 'scope' => null]
+                    ]
+                ]
+            ]);
+
+            $this->createProduct('product_three', [
+                'values' => [
+                    'a_metric' => [
+                        ['data' => ['amount' => '-2.5654', 'unit' => 'KILOWATT'], 'locale' => null, 'scope' => null]
```

Comment:
maybe we should add an integration test where the unit is different from the other to be sure we filter on the base_data, and not on amount